src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -418,6 +424,41 @@ public unsafe void SetResponseHeaders()
             }
         }
 
+        public unsafe void SetResponseTrailers()
+        {
+            HttpResponseTrailers.IsReadOnly = true;
+            foreach (var headerPair in HttpResponseTrailers)
+            {
+                var headerValues = headerPair.Value;
+
+                if (headerValues.Count == 0)
+                {
+                    continue;
+                }
+
+                var headerNameBytes = Encoding.ASCII.GetBytes(headerPair.Key);
+                fixed (byte* pHeaderName = headerNameBytes)
+                {
+                    for (var i = 0; i < headerValues.Count; i++)
+                    {
+                        var isFirst = i == 0;
+
+                        var headerValue = headerValues[i];
+                        if (string.IsNullOrEmpty(headerValue))
+                        {
+                            continue;
```

Comment:
This messes with isFirst. How much do we care about setting replace for the first value? If we do then you need to move isFirst outside the loop and only set it to false after successfully writing a value.